packages/editor/src/store/actions.js

```diff
@@ -330,15 +347,16 @@ export function* savePost( options = {} ) {
 	let path = `/wp/v2/${ postType.rest_base }/${ post.id }`;
 	let method = 'PUT';
 	if ( isAutosave ) {
-		const autoSavePost = yield select(
-			STORE_KEY,
-			'getAutosave',
-		);
+		const currentUser = yield resolveSelect( 'core', 'getCurrentUser' );
+		const currentUserId = currentUser ? currentUser.id : null;
```

Comment:
How well does `getAutosave` and the following logic handle the case that this value would be `null`?

Reply:
I've changed it to `undefined` instead of `null` as `getAutosave` returns early with undefined if no userId is provided.

If `autosave` is also `undefined`, lodash `pick` coerces it back into an object, at which point all of the resulting operations continue to work as expected.

Reply:
> If `autosave` is also `undefined`, lodash `pick` coerces it back into an object, at which point all of the resulting operations continue to work as expected.

There's a fair few number of assumptions here to not have a corresponding unit test or at least some inline code comment outlining the expectations. For example, if someone were to naively replace the `pick` implementation with something different in the future.